src/components/PaginatedTable/TableChunk.tsx

```diff
@@ -1,34 +1,81 @@
 import React from 'react';
 
+import {tableDataApi} from '../../store/reducers/tableData';
+import type {IResponseError} from '../../types/api/error';
 import {getArray} from '../../utils';
+import {useAutoRefreshInterval} from '../../utils/hooks';
+import {ResponseError} from '../Errors/ResponseError';
 
-import {LoadingTableRow, TableRow} from './TableRow';
-import type {Chunk, Column, GetRowClassName} from './types';
+import {EmptyTableRow, LoadingTableRow, TableRow} from './TableRow';
+import type {Column, FetchData, GetRowClassName, SortParams} from './types';
 
-// With original memo generic types are lost
-const typedMemo: <T>(Component: T) => T = React.memo;
+const DEBOUNCE_TIMEOUT = 200;
 
-interface TableChunkProps<T> {
+interface TableChunkProps<T, F> {
     id: number;
-    chunkSize: number;
+    limit: number;
     rowHeight: number;
     columns: Column<T>[];
-    chunkData: Chunk<T> | undefined;
+    filters?: F;
+    sortParams?: SortParams;
     observer: IntersectionObserver;
+    isActive: boolean;
+    tableName: string;
+
+    fetchData: FetchData<T, F>;
     getRowClassName?: GetRowClassName<T>;
+    renderErrorMessage?: (error: IResponseError) => React.ReactNode;
+    onDataFetched: (total: number, found: number) => void;
 }
 
-// Memoisation prevents chunks rerenders that could cause perfomance issues on big tables
-export const TableChunk = typedMemo(function TableChunk<T>({
+export const TableChunk = <T, F>({
     id,
-    chunkSize,
+    limit,
     rowHeight,
     columns,
-    chunkData,
+    fetchData,
+    tableName,
+    filters,
+    sortParams,
     observer,
     getRowClassName,
-}: TableChunkProps<T>) {
+    renderErrorMessage,
+    onDataFetched,
+    isActive,
+}: TableChunkProps<T, F>) => {
     const ref = React.useRef<HTMLTableSectionElement>(null);
+    const [isTimeoutActive, setIsTimeoutActive] = React.useState(true);
+    const timeoutRef = React.useRef(0);
+    const [autoRefreshInterval] = useAutoRefreshInterval();
+
+    const {currentData, error} = tableDataApi.useFetchTableChunkQuery(
+        {
+            offset: id * limit,
+            limit,
+            fetchData: fetchData as FetchData<T, unknown>,
+            filters,
+            sortParams,
+            tableName,
+        },
+        {
+            skip: isTimeoutActive || !isActive,
+            pollingInterval: autoRefreshInterval,
+        },
+    );
+
+    React.useEffect(() => {
+        window.clearTimeout(timeoutRef.current);
```

Comment:
We need to clear timeout when chunk becomes inactive in case when setTimeout callback wasn't triggered
therefore we need to save timeout somewhere outside this useEffect to make window.clearTimeout work

Reply:
return function is called on unmount

we need to call clearTimeout not only when component is unmounted but when it is scrolled through too fast

Reply:
As you may see we moved all logics to TableChunk that is always mounted

Reply:
it is called every time when dependencies changes. In this case when `isActive` changes.

Reply:
searched through docs

>It is important to note, however, that the useEffect cleanup function runs not only when our component wants to unmount, but also right before the execution of the next scheduled effect.

you right